### PR TITLE
fix: only create migrations folder if migrationsList is not used

### PR DESCRIPTION
**Pull request by thomaschaaf**
https://github.com/mikro-orm/mikro-orm/issues/907#issuecomment-713532649

### CODE REVIEW

File: packages/migrations/src/Migrator.ts [hunk 1]
```diff
@@ -38,8 +38,16 @@ export class Migrator {
     });
   }
 
+  async ensurePrerequisites() {
```
Comment (B4nan) [inline packages/migrations/src/Migrator.ts:41]: let's make this private (and move it to the end of the file)

also the optional chaining (on following line) might need `/* istanbul ignore next */` as it usually reports missed branch even if both set/unset paths are taken (due to how it is transpiled)

**Comment by B4nan** [comment]
Looking good, lets wait for the CI.

File: packages/migrations/src/Migrator.ts [hunk 2]
```diff
@@ -39,7 +39,7 @@ export class Migrator {
   }
 
   async createMigration(path?: string, blank = false, initial = false): Promise<MigrationResult> {
-    await ensureDir(Utils.normalizePath(this.options.path!));
+    await this.ensurePrerequisites();
```
Comment (thomaschaaf) [inline packages/migrations/src/Migrator.ts:42]: is it okay that this calls ```this.storage.ensureTable```?

**Comment by thomaschaaf** [comment]
@B4nan I don't think the failing code climate has anything to do with my changes.

**Comment by B4nan** [comment]
Yeah that's fine, some configuration mismatch. Tests are passing and full coverage is remained, that is what is important.

**Pull request merged**